Rubberduck.Core/UI/CodeExplorer/Commands/AddTemplateCommand.cs

```diff
@@ -42,7 +42,8 @@ public AddTemplateCommand(ICodeExplorerAddComponentService addComponentService,
 
         public IEnumerable<ProjectType> AllowableProjectTypes => ApplicableProjectTypes;
 
-        public ComponentType ComponentType => ComponentType.Undefined;
+        //We need a valid component type to add the component in the first place. Then the module content gets overwritten.
+        public ComponentType ComponentType => ComponentType.ClassModule;
```

Comment:
I think we need to pass it from the resource or something; it might not necessarily be a class module in future. At least change the comment to a `TODO` or `FIXME` so it's marked.

Reply:
For VBA, the `ComponentType` used to create the component is entirely irrelevant bacause changing the contents via the file will change it to the appropriate one.

However, I am not sure whether this works for VB6. It would be good if somebody who has the IDE could test whether this works for templates with a different component type.